lib/class_variants/instance.rb

```diff
@@ -46,9 +49,15 @@ def render(slot = :default, **overrides)
       @variants.each do |candidate|
         next unless candidate[:slot] == slot
 
-        if (candidate.keys - [:class, :slot]).all? { |key| criteria[key] == candidate[key] }
-          result << candidate[:class]
+        match = false
+
+        candidate.each_key do |key|
+          next if key == :class || key == :slot
+          match = criteria[key] == candidate[key]
+          break unless match
         end
+
+        result << candidate[:class] if match
```

Comment:
This is the most impactful change. I think this is hit quite a lot when rendering.